pkg/archive/time_unsupported.go

```diff
@@ -1,16 +1,26 @@
-// +build !linux
+// +build !linux,!windows
 
 package archive
 
 import (
-	"syscall"
 	"time"
+
+	"golang.org/x/sys/unix"
 )
 
-func timeToTimespec(time time.Time) (ts syscall.Timespec) {
+func timeToTimespec(time time.Time) (ts unix.Timespec) {
 	nsec := int64(0)
 	if !time.IsZero() {
 		nsec = time.UnixNano()
 	}
-	return syscall.NsecToTimespec(nsec)
+	return unix.NsecToTimespec(nsec)
+}
+
+// Gnu tar and the go tar writer don't have sub-second mtime
+// precision, which is problematic when we apply changes via tar
+// files, we handle this by comparing for exact times, *or* same
+// second count and either a or b having exactly 0 nanoseconds
+func sameFsTimeSpec(a, b unix.Timespec) bool {
```

Comment:
Yeah, this will cover the freebsd / solaris case. It's not ideal because I only really need 2 files instead of 3, but these already existed and seemed like a good place for them anyway.